kinto/core/authentication.py

```diff
@@ -37,6 +44,93 @@ def unauthenticated_userid(self, request):
             return userid
 
 
+@implementer(IAuthenticationPolicy)
+class OpenIDConnectPolicy(base_auth.CallbackAuthenticationPolicy):
+    def __init__(self, realm='Realm'):
+        self.realm = realm
+        self._cache = None
+
+        self._jwt_keys = None
+
+    def unauthenticated_userid(self, request):
+        """Return the FxA userid or ``None`` if token could not be verified.
+        """
+        authorization = request.headers.get('Authorization', '')
+        try:
+            authmeth, payload = authorization.split(' ', 1)
+        except ValueError:
+            return None
+
+        if authmeth.lower() != 'bearer+oidc':
+            return None
+
+        try:
+            # Bearer+OIDC id_token=jwt, access_token=bearer
+            parts = payload.split(',')
+            credentials = {k: v for k, v in [p.strip().split("=") for p in parts]}
+            id_token = credentials["id_token"]
+            access_token = credentials["access_token"]
```

Comment:
Are these lines mandatory in JWT from an OIDC provider?
I've created a non-interactive client on Auth0 dashboard, from the APIs gotten the JWT token using curl, however, I always got HTTP 401 Unauthorized from Kinto.
```
{
  "iss": "https://tenant.auth0.com/",
  "sub": "key@clients",
  "aud": "https://tenant.auth0.com/api/v2/",
  "iat": 1516993206,
  "exp": 1517079606,
  "azp": "gaJ9QUArBJTPpL76SYV8jDhqu9Cztq93",
  "gty": "client-credentials"
}
```
The code above is the Payload extracted from [JWT.io site](https://jwt.io/)

Reply:
Accordingly to section 2 from [Open ID Core](http://openid.net/specs/openid-connect-core-1_0.html#IDToken) the keys required are all right.

Reply:
My bad, I'm testing all day with a non-interactive client from Auth0, the purpose of that is to grant authorization to scripts, in a server-server communication channel.
Tomorrow I will plug Auth0 lock (login) and update this topic with discovers, I'm really happy with how simples this code are at the same time how great are thanks to OIDC discovery.
Thanks to @leplatrem too for spend resources on this code.

Reply:
Status update:
1. First of all, I become testing this in an Ionic project, already configured with Auth0, but when trying to integrate with kinto.js I don't know why, but sadly doesn't work
2. Download quick start from Auth0's GitHub samples, inspecting the code on this branch I've seen that this is relying on python-jose for authentication, specifically to `decode` the `JWT`, also, @leplatrem  fixed an issue and make a PR to address an error on `at_hash`claim being required, but accordingly to OIDC that is not required at all
3. I've forked the @leplatrem repo from python-jose and build the branch with her contributions to generate a wheel (whl) and installed it with pip3, however, I'm always getting an error from Authentication.py 
`Incorrect claims, please check the audience and issuer: **at_hash claim missing from token**.`

I think that the local installation of python-jose (whl) is not being reflected on kinto source, can someone give me a help, I don't know what I have to do to make the decode function of python-jose work with enhanced code from @leplatrem in order to advance in these tests.

The good news is that after all, we already have a sample demonstrating how to integrate OIDC compliant like Auth0 to kinto.